ForumProject/communications/signals.py

```diff
@@ -0,0 +1,32 @@
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.db import models
+from .models import Message, ChatNotification
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from channels.layers import get_channel_layer
+from asgiref.sync import async_to_sync
+
+@receiver(post_save, sender=Message)
+def create_chat_notification(sender, instance, created, **kwargs):
+    if created:
+        ChatNotification.objects.create(
+            recipient=instance.user,
+            message=instance
+        )
+
+@receiver(post_save, sender=Message)
+def send_chat_notification_via_channels(sender, instance, created, **kwargs):
+    if created:
+        chat_notification = ChatNotification.objects.create(
+            recipient=instance.room.online,
+            message=instance
+        )
```

Comment:
Check this case, pls. The handler attempts to assign the recipient of a `ChatNotification` to a queryset `instance.room.online`, which might be incorrect since the recipient field expects a single `User` instance. This may raise an error. You need to iterate over the users in `instance.room.online.all()` and create/send notifications individually.